### PR TITLE
Reset theme buttons

### PR DESCRIPTION
**Pull request by a-wallen**
### Feature Preview

image

fixes https://github.com/AppFlowy-IO/AppFlowy/issues/3085

---

#### PR Checklist

- [X] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [X] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/integration_test/appearance_settings_test.dart [hunk 1]
```diff
@@ -41,5 +42,45 @@ void main() {
 
       expect(find.textContaining('Abel'), findsOneWidget);
     });
+
+    testWidgets('reset the font family', (tester) async {
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/integration_test/appearance_settings_test.dart:46]: the `runner.dart` doesn't include this file.

File: frontend/appflowy_flutter/lib/workspace/application/appearance.dart [hunk 2]
```diff
@@ -15,6 +15,10 @@ import 'package:google_fonts/google_fonts.dart';
 part 'appearance.freezed.dart';
 
 const _white = Color(0xFFFFFFFF);
+const kDefaultFontFamily = 'Poppins';
+const kDefaultThemeMode = ThemeMode.system;
+const kDefaultThemeName = "Default";
```
Comment (MayurSMahajan) [inline frontend/appflowy_flutter/lib/workspace/application/appearance.dart:20]: Where is this `const kDefaultThemeName` used?

File: frontend/appflowy_flutter/lib/workspace/application/appearance.dart [hunk 3]
```diff
@@ -15,6 +15,10 @@ import 'package:google_fonts/google_fonts.dart';
 part 'appearance.freezed.dart';
 
 const _white = Color(0xFFFFFFFF);
+const kDefaultFontFamily = 'Poppins';
```
Comment (MayurSMahajan) [inline frontend/appflowy_flutter/lib/workspace/application/appearance.dart:18]: Can we create a class encapsulating these constants named: `DefaultAppearanceSettings`

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart [hunk 4]
```diff
@@ -394,17 +411,58 @@ class _ThemeFontFamilySettingState extends State<ThemeFontFamilySetting> {
...
+          FlowyIconButton(
+            key: resetButtonKey,
+            width: 24,
+            icon: FlowySvg(
+              name: 'common/recover',
+              color: Theme.of(context).colorScheme.onBackground,
+            ),
+            tooltipText: LocaleKeys.settings_appearance_resetSetting.tr(),
+
+            // fontColor: Theme.of(context).colorScheme.onBackground,
```
Comment (MayurSMahajan) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart:449]: remove comment

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart [hunk 5]
```diff
@@ -394,17 +411,58 @@ class _ThemeFontFamilySettingState extends State<ThemeFontFamilySetting> {
...
+            label,
+            overflow: TextOverflow.ellipsis,
+          ),
+        ),
+        if (trailing != null) ...trailing!,
+        if (onResetRequested != null)
+          FlowyIconButton(
+            key: resetButtonKey,
+            width: 24,
+            icon: FlowySvg(
```
Comment (MayurSMahajan) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart:443]: `icon: const FlowySvg(
              name: 'common/recover',
            ),`
            
I think by default the onBackground color is applied, but please confirm this on your system.

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart [hunk 6]
```diff
@@ -417,31 +475,21 @@ class ThemeSettingDropDown extends StatefulWidget {
...
+        minWidth: 80,
+        maxWidth: 160,
+        maxHeight: 400,
+      ),
+      onClose: widget.onClose,
+      child: FlowyTextButton(
+        widget.currentValue,
+        fontColor: Theme.of(context).colorScheme.onBackground,
+        fillColor: Colors.transparent,
+      ),
```
Comment (MayurSMahajan) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart:492]: I think this file is too big, so can you refactor this and separate these widgets into other files while you are at it?

**Review by MayurSMahajan — CHANGES_REQUESTED** [review]
Some refactoring needed.

File: frontend/appflowy_flutter/lib/workspace/application/appearance.dart (hunk 2 above)
Comment (a-wallen) [reply frontend/appflowy_flutter/lib/workspace/application/appearance.dart:20]: It's just in this file right now. I couldn't figure out a better place to put it. I was waiting for a reviewer to help out ;)

File: frontend/appflowy_flutter/integration_test/appearance_settings_test.dart (hunk 1 above)
Comment (a-wallen) [reply frontend/appflowy_flutter/integration_test/appearance_settings_test.dart:46]: https://github.com/AppFlowy-IO/AppFlowy/pull/3137/commits/db4514a59dff981f1d00efa2727c1aca9b451b5e

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart (hunk 4 above)
Comment (a-wallen) [reply frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart]: https://github.com/AppFlowy-IO/AppFlowy/pull/3137/commits/3ba8be0bd41d3ba18ae837cb46dad998985370fe

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart (hunk 6 above)
Comment (a-wallen) [reply frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart]: https://github.com/AppFlowy-IO/AppFlowy/pull/3137/commits/80efef0277467b5ecc3ddc34615c6853ee7b4604

File: frontend/appflowy_flutter/lib/workspace/application/appearance.dart (hunk 3 above)
Comment (a-wallen) [reply frontend/appflowy_flutter/lib/workspace/application/appearance.dart]: https://github.com/AppFlowy-IO/AppFlowy/pull/3137/commits/fc6516ef7c563f82e0f397a55c459c238e621b9c

File: frontend/appflowy_flutter/lib/workspace/application/appearance.dart (hunk 2 above)
Comment (a-wallen) [reply frontend/appflowy_flutter/lib/workspace/application/appearance.dart]: https://github.com/AppFlowy-IO/AppFlowy/pull/3137/commits/fc6516ef7c563f82e0f397a55c459c238e621b9c

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart (hunk 5 above)
Comment (a-wallen) [reply frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart]: Looks like `svgWidget` uses `Theme.of(context).iconTheme.color`. I'm not sure if it's the same, but I'm going to leave it as is.

**Review by MayurSMahajan — APPROVED** [review]
Looks good to me!

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/color_scheme.dart [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+            bloc: bloc..add(DynamicPluginEvent.load()),
+            buildWhen: (previous, current) => current is Ready,
+            builder: (context, state) {
+              return state.when(
+                uninitialized: () => const SizedBox.shrink(),
+                processing: () => const SizedBox.shrink(),
+                compilationFailure: (message) => const SizedBox.shrink(),
+                deletionFailure: (message) => const SizedBox.shrink(),
+                deletionSuccess: () => const SizedBox.shrink(),
+                compilationSuccess: () => const SizedBox.shrink(),
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/color_scheme.dart:108]:
```suggestion
              return state.maybeWhen(
                orElse: () => const SizedBox.shrink(),
```

**Review by Xazin — APPROVED** [review]

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/font_family_setting.dart [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+  static Key resetButtonkey = const Key('FontFamilyResetButton');
+  static Key popoverKey = const Key('FontFamilyPopover');
+
+  @override
+  State<ThemeFontFamilySetting> createState() => _ThemeFontFamilySettingState();
+}
+
+class _ThemeFontFamilySettingState extends State<ThemeFontFamilySetting> {
+  final List<String> availableFonts = GoogleFonts.asMap().keys.toList();
+  final ValueNotifier<String> query = ValueNotifier('');
```
[the remark this answers is not in the document]
Comment (a-wallen) [reply frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/font_family_setting.dart:33]: I'd address this in a different PR.

**Comment by a-wallen** [comment]
image

Icon looks better now

File: frontend/appflowy_flutter/lib/workspace/application/appearance.dart [hunk 9]
```diff
@@ -43,13 +44,21 @@ class AppearanceSettingsCubit extends Cubit<AppearanceSettingsState> {
     emit(state.copyWith(appTheme: await AppTheme.fromName(themeName)));
   }
 
+  /// Reset the current user selected theme back to the default
+  Future<void> resetTheme() =>
+      setTheme(DefaultAppearanceSettings.kDefaultFontFamily);
```
[the remark this answers is not in the document]
Comment (a-wallen) [reply frontend/appflowy_flutter/lib/workspace/application/appearance.dart]: https://github.com/AppFlowy-IO/AppFlowy/pull/3137/commits/839df0673002c75182dd328bda9a7b67208617c3

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/color_scheme.dart (hunk 7 above)
Comment (a-wallen) [reply frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/color_scheme.dart]: https://github.com/AppFlowy-IO/AppFlowy/pull/3137/commits/4757fd2d1c0b203b92be4b91e98713e4b451b86a

**Pull request merged**